doxygen/contributor_help_pages/require_meta.md

```diff
@@ -0,0 +1,191 @@
+
+```cpp
+ // Works for non standard vector types
+ template <typename T, require_not_std_vector_t<T>* = nullptr>
+ auto add(T&& x, T&& y) { return x + y; }
+```
+
+- `require_all_*_t` : Takes a parameter pack of types to enable if all types
+satisfy the check.
+
+```cpp
+ // Works if T1 and T2 are complex
+ template <typename T1, typename T2,
+   require_all_complex_t<T1, T2>* = nullptr>
+ auto add(T1&& x, T2&& y) { return x + y; }
+```
+
+- `require_any_*_t` : Takes a parameter pack of types to enable if any of
+the types satisfy the check.
+
+```cpp
+ // Works if either T1 or T2 enherit from EigenBase
+ template <typename T1, typename T2, require_any_eigen_t<T1, T2>* = nullptr>
+ auto add(T1&& x, T2&& y) { return x + y; }
+```
+
+- `require_not_any_*_t` : Takes a parameter pack of types to enable if any
+one of the types are not satisfied.
+
+```cpp
+ // Works if either neither T1 or T2 are arithmetic
+ template <typename T1, typename T2,
+   require_not_any_eigen_row_vector_t<T1, T2>* = nullptr>
+ auto add(T1 x, T2 y) { return x + y; }
+```
+
+- `require_not_all_*_t` : Takes a parameter pack of types to enable if all
+of the types are not satisfied.
+
+```cpp
+ // Works if neither T1 and T2 are arithmetic
+ template <typename T1, typename T2,
+   require_not_all_arithmetic_t<T1, T2>* = nullptr>
+ auto add(T1 x, T2 y) { return x + y; }
+```
+
+ `std::vector` and `Eigen` types have additional requires to
+detect if the @ref stan::value_type (the first underlying type) or the
+ @ref stan::scalar_type
+(the containers underlying scalar type) satisfy a condition to enable a class
+or function.
+
+ The container `requires` have an ending at their signature of _vt and _st to symbolize whether you want to inspect the @ref stan::value_type or @ref stan::scalar_type. A function
+that accepts eigen matrices with floating point value types can be defined as
+
+```cpp
+ template <typename Mat1, typename Mat2,
+   require_all_eigen_vt<std::is_floating_point, Mat1, Mat2>* = nullptr>
+ auto add(Mat1&& A, Mat2&& B) { return A + B;}
+```
+  A function that accepts standard vectors of Eigen vectors whose scalar type
+is @ref stan::math::var types can be defined as
+
+```cpp
+ template <typename Vec1, typename Vec2,
+   require_all_std_vector_vt<is_eigen_vector, Vec1, Vec2>* = nullptr,
+   require_all_std_vector_st<is_var, Vec1, Vec2>* = nullptr>
+ auto add(Vec1&& A, Vec2&& B) {
+   std::vector<decltype<A[0] + B[0]>> return_vec;
+   std::transform(A.begin(), A.end(), B.begin(), return_vec.begin(),
+     [](auto&& x, auto&& y) {
+         return x + y;
+     });
+   return return_vec;
+ }
+```
+
+ There are also requires for generically checking if a type's @ref stan::value_type or
+  @ref stan::scalar_type is correct. To differentiate them from the Eigen and Standard
```

Comment:
Standard -> standard library

comes -> come (it's plural)

Reply:
Fixed